app/Http/Controllers/DemandController.php

```diff
@@ -0,0 +1,334 @@
+        $pid = $request->query->get('pid');
+        $tid = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
+        $payTo = $request->query->get('pto');
+
+        $keywords = json_decode(Utils::UrlSafeBase64Decode($request->query->get('k')), true);
+
+        $lid = $request->query->get('lid');
+        if ($lid) {
+            $log = EventLog::find($lid);
+        }
+        if (empty($log)) {
+            $log = new EventLog();
+        }
+        $log->publisher_event_id = $pid;
+
+        $log->cid = $cid;
+        $log->banner_id = $id;
+        $log->pay_to = $payTo;
+        $log->tid = $tid;
+        $log->ip = $logIp;
+        $log->their_context = Utils::getImpressionContext($request);
+        $log->event_type = "view";
+        $log->their_userdata=$keywords;
+
+        $log->save();
+
+        $aduser_endpoint = config('app.aduser_endpoint');
+
+        if ($aduser_endpoint) {
+            $iid = $request->query->get('iid') ?: Utils::createTrackingId($this->getParameter('secret'));
+            $backUrl = route('log-keywords', [  'iid' => $iid,
+              'log_id' => $log->id,
+              'r' => $url
+            ]);
+
+            $response = new RedirectResponse($aduser_endpoint . '/setimg/' . $iid . '?r='. Utils::UrlSafeBase64Encode($backUrl));
+        } else {
+            throw new Exception('ADAPY');
+
+            $response = new Response();
+
+            //transparent 1px gif
+            $response->setContent(base64_decode('R0lGODlhAQABAIABAP///wAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=='));
+            $response->headers->set('Content-Type', 'image/gif');
+
+            $adpayService = $this->container->has('adpay') ? $this->container->get('adpay') : null;
+            $adpayService instanceof Adpay;
+
+            if ($adpayService) {
+                $adpayService->addEvents([
+                  $log->getAdpayJson(),
+                ]);
+            }
+        }
+
+        return $response;
+    }
+
+    public function logKeywords(Request $request, $log_id)
+    {
+        $url = Utils::UrlSafeBase64Decode($request->query->get('r'));
+
+        // GET kewords from aduser
+        $impressionId = $request->query->get('iid');
+        $aduser_endpoint = config('app.aduser_endpoint');
+        $userdata = ($aduser_endpoint && $impressionId) ? json_decode(file_get_contents("{$aduser_endpoint}/get/{$impressionId}"), true) : [];
+
+        $log = EventLog::find($log_id);
+        if (!empty($log)) {
+            $log->our_userdata =$userdata['keywords'];
+            $log->human_score = $userdata['human_score'];
+            $log->user_id = $userdata['user_id'];
+            $log->save();
+        }
+
+        $url = Utils::addUrlParameter($url, 's', parse_url($aduser_endpoint, PHP_URL_HOST));
+        $url = Utils::addUrlParameter($url, 'k', Utils::UrlSafeBase64Encode(json_encode($userdata['keywords'])));
+
+        $response = new RedirectResponse($url);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)